connectors/tests/ftest.sh

```diff
@@ -37,34 +37,54 @@ fi
 $PYTHON -m pip install -r $NAME/requirements.txt
 $PYTHON fixture.py --name $NAME --action setup
 $PYTHON fixture.py --name $NAME --action start_stack
-$ROOT_DIR/bin/fake-kibana --index-name search-$NAME --service-type $NAME --debug --filtering $NAME/filtering.json
-$PYTHON fixture.py --name $NAME --action load
-
-if [[ $PERF8 == "yes" ]]
-then
-    if [[ $PLATFORM == "darwin" ]]
-    then
-      $PERF8_BIN --refresh-rate $REFRESH_RATE -t $ROOT_DIR/perf8-report-$NAME --asyncstats --memray --psutil -c $ELASTIC_INGEST --one-sync --sync-now --debug
-    else
-      $PERF8_BIN --refresh-rate $REFRESH_RATE -t $ROOT_DIR/perf8-report-$NAME --asyncstats --memray --psutil -c $ELASTIC_INGEST --one-sync --sync-now --debug
-    fi
-else
-    $ELASTIC_INGEST --one-sync --sync-now --debug
-fi
-
-$PYTHON fixture.py --name $NAME --action remove
-
-$ELASTIC_INGEST --one-sync --sync-now --debug
 
+run_test()
+{
+  # check if path to a filtering.json file is present
+  if [ -n "${1-}" ];
+  then
+    $ROOT_DIR/bin/fake-kibana --index-name search-$NAME --service-type $NAME --debug --filtering $NAME/$1
+  else
+    $ROOT_DIR/bin/fake-kibana --index-name search-$NAME --service-type $NAME --debug
+  fi
+
+  $PYTHON fixture.py --name $NAME --action load
+
+  if [[ $PERF8 == "yes" ]]
+  then
+      if [[ $PLATFORM == "darwin" ]]
+      then
+        $PERF8_BIN --refresh-rate $REFRESH_RATE -t $ROOT_DIR/perf8-report-$NAME --asyncstats --memray --psutil -c $ELASTIC_INGEST --one-sync --sync-now --debug
+      else
+        $PERF8_BIN --refresh-rate $REFRESH_RATE -t $ROOT_DIR/perf8-report-$NAME --asyncstats --memray --psutil -c $ELASTIC_INGEST --one-sync --sync-now --debug
+      fi
+  else
+      $ELASTIC_INGEST --one-sync --sync-now --debug
+  fi
+
+  $PYTHON fixture.py --name $NAME --action remove
+
+  $ELASTIC_INGEST --one-sync --sync-now --debug
+
+
+  if [[ "$DATA_SIZE" == 'small' ]]; then
+     SIZE=750
+  elif [[ "$DATA_SIZE" == 'medium' ]]; then
+     SIZE=1500
+  else
+     SIZE=3000
+  fi
+
+  $PYTHON $ROOT_DIR/scripts/verify.py --index-name search-$NAME --service-type $NAME --size $SIZE
+}
+
+# run test once without filtering
+run_test
+
+for filtering in ./filtering*.json
+do
+  run_test $filtering
```

Comment:
Somehow it feels that `verify` can be different per filtering setup too - different number of entries should be present or some other criteria checked. Are you planning to add it in upcoming PRs?

Reply:
I've created an issue an hour ago about addressing exactly this issue (and some others). I would've probably the time before 8.8 kick-off implementing the proposal...I'll double check, if there's room this week to address those. Feel free to comment on it.